webapp/src/snippetBuilder.tsx

```diff
@@ -0,0 +1,398 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+import * as React from "react";
+import * as data from "./data";
+import * as sui from "./sui";
+import * as md from "./marked";
+import * as compiler from './compiler';
+
+type ISettingsProps = pxt.editor.ISettingsProps;
+
+export interface SnippetBuilderState {
+    visible?: boolean;
+    tsOutput?: string;
+    projectView?: pxt.editor.IProjectView;
+    answers?: any; // Will be typed once more clearly defined
+    currentQuestion: number;
+    defaults: any; // Will be typed once more clearly defined
+    mainWorkspace?: Blockly.Workspace;
```

Comment:
does not belong to the state

Reply:
I'm now passing this down as the parent prop.

Reply:
Passing this down as a prop now.